packages/amplify-provider-awscloudformation/src/utils/upload-auth-trigger-template.ts

```diff
@@ -0,0 +1,39 @@
+import { $TSContext, $TSObject, FeatureFlags, JSONUtilities, pathManager } from 'amplify-cli-core';
+import { S3 } from '../aws-utils/aws-s3';
+import path from 'path';
+
+export const AUTH_TRIGGER_TEMPLATE = 'auth-trigger-cloudformation-template.json';
+export const AUTH_TRIGGER_STACK = 'AuthTriggerCustomLambdaStack';
+
+export async function uploadAuthTriggerTemplate(context: $TSContext) {
+  if (!FeatureFlags.getBoolean('auth.breakCircularDependency')) {
+    return {};
+  }
+  const category = 'auth';
+  let { amplifyMeta } = context.amplify.getProjectDetails();
+  const authResource = amplifyMeta?.auth ?? {};
+  const authTriggerCfnFilePath = path.join(pathManager.getBackendDirPath(), category, Object.keys(authResource)[0], AUTH_TRIGGER_TEMPLATE);
+  try {
+    const cfnObject: $TSObject = JSONUtilities.readJson(authTriggerCfnFilePath);
+    const S3_UPLOAD_PATH = path.join(category, AUTH_TRIGGER_TEMPLATE);
+
+    return {
+      AuthTriggerTemplateURL: await uploadCfnToS3(context, S3_UPLOAD_PATH, cfnObject),
+    };
+  } catch {
+    return {
```

Comment:
seems like there are 2 different failures being caught here. 1 is the trigger template doesn't exist and 2 is it failed to upload to S3. Seems like for 2 we would want to throw an error